tests/shell/src/main.c

```diff
@@ -295,6 +295,27 @@ static void test_cmd_select(void)
 	test_shell_execute_cmd("on", -ENOEXEC);
 }
 
+static void test_set_root_cmd(void)
+{
+	int err;
+
+	test_shell_execute_cmd("shell colors on", 0);
+	err = shell_set_root_cmd("__shell__");
+	zassert_equal(err, -EINVAL, "Unexpected error %d", err);
+
+	err = shell_set_root_cmd("shell");
```

Comment:
Will this new command be available from the command line?

Reply:
well, `select` command does this. It is just API to do the same from the code. The primary use case is the AT modem which is using shell but talks to external tool which does not understand other commands than `AT`. In case of such test application, `shell_set_root_cmd("at")` is called during init to start already in "at" mode.